themes-default/slim/src/components/subtitle-search.vue

```diff
@@ -0,0 +1,258 @@
+                    <a :title="`Download ${props.row.hearing_impaired ? 'hearing impaired ' : ' '} subtitle: ${props.row.filename}`" @click="pickSubtitle(props.row.id)">
+                        <img v-if="props.row.hearing_impaired" src="images/hearing_impaired.png" width="16" height="16"/>
+                        <span class="subtitle-name">{{props.row.filename}}</span>
+                        <img v-if="props.row.sub_score >= props.row.min_score" src="images/save.png" width="16" height="16"/>
+                    </a>
+                </span>
+                <span v-else-if="props.column.field == 'download'">
+                    <a :title="`Download ${props.row.hearing_impaired ? 'hearing impaired ' : ' '} subtitle: ${props.row.filename}`" @click="pickSubtitle(props.row.id)">
+                        <img src="images/download.png" width="16" height="16"/>
+                    </a>
+                </span>
+                <span v-else>
+                    {{props.formattedRow[props.column.field]}}
+                </span>
+            </template>
+        </vue-good-table>
+    </td>
+</tr>
+</template>
+<script>
+
+import { mapState } from 'vuex';
+import { VueGoodTable } from 'vue-good-table';
+
+export default {
+    name: 'subtitle-search',
+    components: {
+        VueGoodTable
+    },
+    props: {
+        show: {
+            type: Object,
+            required: true
+        },
+        season: {
+            type: [String, Number],
+            required: true
+        },
+        episode: {
+            type: [String, Number],
+            required: true
+        }
+    },
+    data() {
+        return {
+            columns: [{
+                label: 'Filename',
+                field: 'filename'
+            }, {
+                label: 'Language',
+                field: 'lang'
+            }, {
+                label: 'Provider',
+                field: 'provider'
+            }, {
+                label: 'Score',
+                field: 'score',
+                type: 'number'
+            }, {
+                label: 'Sub Score',
+                field: 'sub_score',
+                type: 'number'
+            }, {
+                label: 'Missing Matches',
+                field: rowObj => {
+                    if (rowObj.missing_guess) {
+                        return rowObj.missing_guess.join(', ');
+                    }
+                },
+                type: 'array'
+            }, {
+                label: 'Download',
+                field: 'download'
+            }],
+            subtitles: [],
+            displayQuestion: false,
+            loading: false,
+            loadingMessage: ''
+        };
+    },
+    computed: {
+        ...mapState({
+            config: state => state.config
+        })
+    },
+    mounted() {
+        this.displayQuestion = true;
+    },
+    destroyed() {
+        // Remove the element from the DOM
+        this.$el.parentNode.removeChild(this.$el);
```

Comment:
😬 
Okay... two things.
1. **I think** that when a component gets destroyed, it means it removes anything it rendered.
	So... you shouldn't need to do that. If you encountered an issue with that logic, let me know and we'll try to work around it.

2. I would not do it (hide/remove the component) this way.
	Vue uses a VDOM, as in *Virtual* DOM.
	If you remove an element that is tied to a component or a state, my guess is as soon as the state changes, or when the component needs to update the DOM, it will just re-create it.

	My suggestion?
	This component should emit an event to let its parent know that it is no longer "needed",
	and the parent component should use `v-if="subtitleSelectionVisible"` on this component.
	This will trigger an automatic (and more reliable IMO) destruction of this component.
	Plus it makes this component more detached from its parent, in the sense that it doesn't manipulate it.

P.S.
Actually I look at this and I think of how complex the DOM manipulation is on [`ajax-episode-subtitles.js`](https://github.com/pymedusa/Medusa/blob/develop/themes-default/slim/static/js/ajax-episode-subtitles.js)/[`ajax-episode-search.js`](https://github.com/pymedusa/Medusa/blob/b1a1a86ed02787ee1a365b2356c5031b1bb582fc/themes-default/slim/static/js/ajax-episode-search.js), and we don't want that, do we? 😄

Reply:
it's really needed. I can call $destory() as much as I want. But it won't remove the node.
I really need to have it removed. I tested, it won't recreate it.